toqito/matrix_ops/commutant.py

```diff
@@ -0,0 +1,104 @@
+"""Module for computing the commutant of a set of matrices."""
+
+import numpy as np
+from scipy.linalg import null_space
+
+
+def commutant(A: np.ndarray | list[np.ndarray]) -> list[np.ndarray]:
+    r"""Compute an orthonormal basis for the commutant algebra.
+
+    Given a set of matrices :math:`A`, this function determines an orthonormal basis
+    (with respect to the Hilbert-Schmidt inner product) for the algebra of matrices
+    that commute with every matrix in :math:`A`.
+
+    The commutant of a set of matrices consists of all matrices that commute
+    with every matrix in the set. Formally, given a matrix :math:`A \in \mathbb{C}^{n \times n}`,
+    a matrix :math:`X \in \mathbb{C}^{n \times n}` belongs to the commutant if:
+
+    .. math:: A X = X A \quad \forall A \in \mathcal{A}.
+
+    This can be rewritten as:
+
+    .. math::
+        (A \otimes I - I \otimes A^T) \text{vec}(X) = 0.
+
+    where :math:`\text{vec}(X)` denotes the vectorization of :math:`X`, which arranges its columns
+    into a single column vector. The null space of this equation provides a basis for the commutant
+    :cite:`johnston2016qetlab`.
```

Comment:
```suggestion
    :cite:`QETLAB_link`.
```

Reply:
FYI I made this change here because there was an error due to an invalid bib key.